src/openzaak/utils/permissions.py

```diff
@@ -138,13 +145,67 @@ def has_object_permission(self, request: Request, view, obj) -> bool:
         if not self.permission_fields:
             return request.jwt_auth.has_auth(scopes_required, component)
 
-        main_resource = self.get_main_resource()
+        main_resource = self.get_main_resource(self.main_resource)
 
         if view.__class__ is main_resource:
             main_object = obj
         else:
             main_object = self.get_main_object(obj, view.permission_main_object)
 
-        main_object_data = self.format_data(main_object, request)
-        fields = self.get_fields(main_object_data)
+        main_object_data = self.format_data(main_object, request, main_resource)
+        fields = self.get_fields(main_object_data, self.permission_fields)
         return request.jwt_auth.has_auth(scopes_required, component, **fields)
+
+
+class MultipleObjectsAuthRequired(AuthRequired):
+    permission_fields: dict
```

Comment:
Let's make these types more specific
```suggestion
    permission_fields: dict[str, Sequence[str]]
```